pkg/podexecutor/pebinary.go

```diff
@@ -0,0 +1,305 @@
+// +build windows
+
+package podexecutor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/k3s-io/helm-controller/pkg/generated/controllers/helm.cattle.io"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/Microsoft/hcsshim"
+
+	"golang.org/x/sys/windows/registry"
+
+	"github.com/rancher/k3s/pkg/cli/cmds"
+	daemonconfig "github.com/rancher/k3s/pkg/daemons/config"
+	"github.com/rancher/k3s/pkg/daemons/executor"
+	"github.com/rancher/rke2/pkg/bootstrap"
+	"github.com/rancher/rke2/pkg/images"
+	win "github.com/rancher/rke2/pkg/windows"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+)
+
+var (
+	ssldirs = []string{
+		"/etc/ssl/certs",
+		"/etc/pki/tls/certs",
+		"/etc/ca-certificates",
+		"/usr/local/share/ca-certificates",
+		"/usr/share/ca-certificates",
+	}
+)
+
+type PEBinary struct {
+	ManifestsDir    string
+	ImagesDir       string
+	Resolver        *images.Resolver
+	CloudProvider   *CloudProviderConfig
+	CISMode         bool
+	DataDir         string
+	AuditPolicyFile string
+	KubeletPath     string
+	DisableETCD     bool
+	IsServer        bool
+	cniConig        *win.CniConfig
+	cni             win.Cni
+}
+
+type CloudProviderConfig struct {
+	Name string
+	Path string
+}
+
+const (
+	windowsReleaseId   = "1903"
```

Comment:
Nit but part of the Go style guide expectations.
```suggestion
	windowsReleaseID   = "1903"
```

Reply:
I think these can be dropped entirely, as hcsshim has a function for checking if DSR is enabled that doesn't require manually poking release IDs out of the registry